### PR TITLE
chore(bot): add bot app registration error telemetry

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Add bot app registration error telemetry
Detail: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/14549235

### CODE REVIEW

**Review by yukun-dong — APPROVED** [review]

**Review by yukun-dong — DISMISSED** [review]

**Review by a1exwang — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/bot/constants.ts [hunk 1]
```diff
@@ -125,6 +125,7 @@ export class ErrorNames {
   public static readonly MSG_ENDPOINT_UPDATING_ERROR = "MessageEndpointUpdatingError";
   public static readonly COMMAND_EXECUTION_ERROR = "CommandExecutionError";
   public static readonly CALL_APPSTUDIO_API_ERROR = "CallAppStudioAPIError";
+  public static readonly API_ERROR = "ApiError";
```
Comment (eriolchan) [inline packages/fx-core/src/plugins/resource/bot/constants.ts:128]: It seems that the telemetry is tracking app studio api error, can it reuse the above CALL_APPSTUDIO_API_ERROR ?

File: packages/fx-core/src/plugins/resource/bot/appStudio/appStudio.ts [hunk 2]
```diff
@@ -176,7 +186,7 @@ export class AppStudio {
             }
           },
           true
-        );
+        ).catch((e) => sendAppStudioApiTelemetry(telemetryReporter, e));
```
Comment (eriolchan) [inline packages/fx-core/src/plugins/resource/bot/appStudio/appStudio.ts:189]: Is it possible to do this catch and send telemetry outside the appStudio.ts? It looks too intrusive to the existing code to pass telemetryReport to each function call, which make code not clean. There is already a guideline in Bot plugin on how to handle exception and telemetry in a unified way, @IvanJobs can you help review this PR and suggest the right way to add telemetry for appStudio call?

Comment (IvanJobs) [reply packages/fx-core/src/plugins/resource/bot/appStudio/appStudio.ts:189]: @XiaofuHuang I agree with @eriolchan on the intrusive part.
According to the original design, for example, https://github.com/OfficeDev/TeamsFx/blob/60d497a45bf7987c3037a0c06b131393ed644f37/packages/fx-core/src/plugins/resource/bot/index.ts#L100, all exceptions are supposed to be caught on upper layer - `index.ts` which will significantly simplify the logic, provide single source of truth and reduce a lot of scattered code like what it is in the PR.

How about moving the logic on top layer? We don't want `context` or `telemetryReporter` being passed for all lower layers functions like this.

Comment (XiaofuHuang) [reply packages/fx-core/src/plugins/resource/bot/appStudio/appStudio.ts]: Thx. Moved it to telemetryReporter

File: packages/fx-core/src/plugins/resource/bot/constants.ts (hunk 1 above)
Comment (XiaofuHuang) [reply packages/fx-core/src/plugins/resource/bot/constants.ts]: Removed the `ApiError` error event.

**Review by IvanJobs — APPROVED** [review]
lgtm

**Pull request merged**